### PR TITLE
developed api to set the power limit

### PR DESCRIPTION
**Pull request by PoornachandraraoD**
developed api to set the power limit.

### CODE REVIEW

File: examples/test_hardware_management.py [hunk 1]
```diff
@@ -92,3 +92,33 @@ def test_get_multi_power_usage():
...
+    """
+            Method to get power usage.
+            :return: power usage.
+            usage : pytest test_hardware_management.py::test_get_power_usage
+    """
+    bool_resp, result = ob.set_power_limit(power_limit=540)
+    print(result)
+
+'''
+def test_set_power_usage_value_1():
```
Comment (sathishpos) [inline examples/test_hardware_management.py]: Why do you need 2 functions? Can't you add in an argument?

File: examples/test_hardware_management.py [hunk 2]
```diff
@@ -92,3 +92,33 @@ def test_get_multi_power_usage():
     html_tb = gr.create_html_table(power, headers)
     gr.generate_hardware_report(html_tb)
 
+def test_set_power_usage_value():
```
Comment (deeptis13) [inline examples/test_hardware_management.py:95]: method name is "test_set_power_usage_value" and doc string is mentioning that this API is to get the power usage. Please correct it

File: examples/test_hardware_management.py (hunk 1 above)
Comment (deeptis13) [inline examples/test_hardware_management.py]: Same error. Please correct the doc string/method name accordingly

File: examples/test_hardware_management.py [hunk 3]
```diff
@@ -92,3 +92,33 @@ def test_get_multi_power_usage():
...
+            usage : pytest test_hardware_management.py::test_get_power_usage
+    """
+    bool_resp, result = ob.set_power_limit(power_limit=540)
+    print(result)
+
+'''
+def test_set_power_usage_value_1():
+    """
+            Method to get power usage.
+            :return: power usage.
```
Comment (deeptis13) [inline examples/test_hardware_management.py]: Please correct the doc string. Also provide the method name that can reflect the actual functionality of the API.

File: examples/test_hardware_management.py [hunk 4]
```diff
@@ -92,3 +92,33 @@ def test_get_multi_power_usage():
...
+            usage : pytest test_hardware_management.py::test_get_power_usage
+    """
+    bool_resp, result = ob.set_power_limit_correction_time(power_limit_correction_time=300)
+    print(result)
+
+def test_set_power_usage_value_2():
+    """
+            Method to get power usage.
+            :return: power usage.
```
Comment (deeptis13) [inline examples/test_hardware_management.py]: Please correct the doc string. Provide method name based on the functionality. It should be user friendly

File: lib/redfish_api.py [hunk 5]
```diff
@@ -69,19 +68,30 @@ def __init__(self):
                         password=self.config_dict["systems"][host]["password"])
                     self.REDFISH_OBJ.login(auth="basic")
 
-                    self.redfish_version = (self.REDFISH_OBJ.get(self.config_dict["base_url"], None)).dict["RedfishVersion"]
-                    print("Version ", self.redfish_version)
```
Comment (deeptis13) [inline lib/redfish_api.py:73]: why this code is removed? please help me to understand the reason.

File: lib/redfish_api.py [hunk 6]
```diff
@@ -69,19 +68,30 @@ def __init__(self):
...
+                    else:
+                        self.system_manufacturer = (self.REDFISH_OBJ.get(self.config_dict["system_url"], None)).dict[
+                            "Manufacturer"]
+                        logger.info(self.system_manufacturer)
                         self.system_model = (self.REDFISH_OBJ.get(self.config_dict["system_url"], None)).dict["Model"]
+                        logger.info(self.system_model)
                     self.REDFISH_OBJ_list.append(self.REDFISH_OBJ)
             except:
-                traceback.print_exc()
+                # traceback.print_exc()
```
Comment (deeptis13) [inline lib/redfish_api.py]: Please provide reason to keep it as commented

File: lib/redfish_api.py [hunk 7]
```diff
@@ -263,28 +274,15 @@ def get_openapi_from_redfish_org(self):
...
-
         try:
-            #if len(self.openapi_dict) > 0:
-            #    pass
-            #    #return
-            filename = self.config_dict["filename"]
-            if os.path.exists(filename) != True:
-                print("Not exists", self.config_dict["openapi_url"])
-                result = requests.get(self.config_dict["openapi_url"])
-                if result.status_code == 200:
```
Comment (deeptis13) [inline lib/redfish_api.py]: Are we changing the logic here? 
Previously we were checking if open api yaml file exists or not , if not then download the file.

File: lib/redfish_api.py [hunk 8]
```diff
@@ -705,37 +703,59 @@ def get_storage_inventory(self) -> (bool, list):
             logger.error("error_msg: {}".format(e))
             return False, storage_details
 
-    '''def set_power_limit(self,isenable,power_limit):
-        isenable=bool(isenable)
+    def set_power_limit(self,power_limit=500):
```
Comment (deeptis13) [inline lib/redfish_api.py:729]: Please provide docstring

**Review by deeptis13 — CHANGES_REQUESTED** [review]

File: examples/test_hardware_management.py (hunk 2 above)
Comment (PremchandJujjuri) [reply examples/test_hardware_management.py:95]: Addressed the review comment

File: examples/test_hardware_management.py (hunk 1 above)
Comment (PremchandJujjuri) [reply examples/test_hardware_management.py]: Addressed the review comment

File: examples/test_hardware_management.py (hunk 3 above)
Comment (PremchandJujjuri) [reply examples/test_hardware_management.py]: Addressed the review comment

File: examples/test_hardware_management.py (hunk 4 above)
Comment (PremchandJujjuri) [reply examples/test_hardware_management.py]: Addressed the review comment

File: lib/redfish_api.py (hunk 5 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py:73]: Added back in latest commit

File: lib/redfish_api.py (hunk 6 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py]: Uncommented in latest commit

File: lib/redfish_api.py (hunk 8 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py:729]: Addressed the review comment

File: examples/test_hardware_management.py (hunk 1 above)
Comment (PremchandJujjuri) [reply examples/test_hardware_management.py]: Addressed the review comment, Modified the API name as per the functionality

File: lib/redfish_api.py (hunk 7 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py]: Added back the logic in latest commit

**Review by deeptis13 — APPROVED** [review]

**Comment by PremchandJujjuri** [comment]
Closing this pull request

**Pull request closed without merging**